test/spec/controllers/certificatesmodal.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+/* jshint expr: true */
+/* global sinon: false */
+
+describe('Controller: CertificatesModalCtrl', function () {
+  // load the controller's module
+  beforeEach(module('openhimConsoleApp'));
+
+  // setup config constant to be used for API server details
+  beforeEach(function(){
+    module('openhimConsoleApp', function($provide){
+      $provide.constant('config', { 'protocol': 'https', 'host': 'localhost', 'port': 8080, 'title': 'Title', 'footerTitle': 'FooterTitle', 'footerPoweredBy': 'FooterPoweredBy' });
+    });
+  });
+
+  var scope, createController, httpBackend;
+
+  // Initialize the controller and a mock scope
+  beforeEach(inject(function ($controller, $rootScope, $httpBackend) {
+
+    httpBackend = $httpBackend;
+
+    $httpBackend.when('POST', new RegExp('.*/certificates')).respond(
+      {
+        certificate: '---BEGIN CERTIFICATE ---',
+        key: '---BEGIN KEY ---'
+      }
+    );
+
+    scope = $rootScope.$new();
+    var modalInstance = sinon.spy();
+
+    createController = function () {
+      var cert;
+      cert = {
+        $save: sinon.spy(),
+        $update: sinon.spy()
+      };
+      return $controller('CertificatesModalCtrl', {
+        $scope: scope,
+        $modalInstance: modalInstance,
+        cert: cert
+      });
+    };
+  }));
+
+  afterEach(function() {
+    httpBackend.verifyNoOutstandingExpectation();
+    httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('should create a new certificate if this is not an update', function () {
```

Comment:
I copied these test fixtures from the clients modal tests because they are similar.  That must have lingered from there, am deleting all the unnecessary code.